tests/testImageMapReduce.py

```diff
@@ -167,71 +166,113 @@ def testExampleTaskWithOverlaps(self):
 
     def testGridValidity(self):
-        """!Test sample grids with various spacings and sizes and other options.
+        """Test sample grids with various spacings and sizes and other options.
 
         Try to see if we can break it.
         """
-        config = TestImageMapReduceConfig()
-        config.reducerSubtask.reduceOperation = 'copy'
         expectedVal = 1.
+        n_tests = 0
+
+        for reduceOp in ('copy', 'average'):
+            for gstepx in range(11, 2, -3):
+                for gsizex in gstepx + np.array([0, 1, 2]):
+                    for gstepy in range(11, 2, -3):
+                        for gsizey in gstepy + np.array([0, 1, 2]):
+                            config = AddAmountImageMapReduceConfig()
+                            config.reducerSubtask.reduceOperation = reduceOp
+                            n_tests += 1
+                            self._testGridValidity(config, gstepx, gsizex, gstepy, gsizey, expectedVal)
+        print("Ran a total of %d grid validity tests." % n_tests)
+
+    def _testGridValidity(self, config, gstepx, gsizex, gstepy, gsizey, expectedVal=1.):
+        """Method to test the grid validity given an input config.
+
+        Here we also iterate over scaleByFwhm in (True, False) and
+        ensure that we get more `boxes` when `scaleByFwhm=False` than
+        vice versa.
+
+        Parameters
+        ----------
+        config : `ipDiffim.AddAmountImageMapReduceConfig`
+            input AddAmountImageMapReduceConfig
+        gstepx : `float`
+            grid x-direction step size
+        gsizex : `float`
+            grid x-direction box size
+        gstepy : `float`
+            grid y-direction step size
+        gsizey : `float`
+            grid y-direction box size
+        expectedVal : `float`
+            float to add to exposure (to compare for testing)
+        """
         config.mapperSubtask.addAmount = expectedVal
-
-        for scaleByFwhm in (False, True):
+        lenBoxes = [0, 0]
+        for scaleByFwhm in (True, False):
             config.scaleByFwhm = scaleByFwhm
-            for gsx in range(51, 1, -10):
-                config.gridStepX = config.gridSizeX = gsx
-                for gsy in range(51, 1, -10):
-                    config.gridStepY = config.gridSizeY = gsy
-
-                    task = ImageMapReduceTask(config)
-                    boxes = task._generateGrid(self.exposure)
-                    if len(boxes[0]) > 1000:  # bypass to prevent slow testing
-                        continue
-                    newExp = task.run(self.exposure)
-                    newMI = newExp.getMaskedImage()
-                    newArr = newMI.getImage().getArray()
-                    mi = self.exposure.getMaskedImage()
-
-                    mn = newArr.mean()
-                    self.assertClose(mi.getImage().getArray().mean(), mn - expectedVal)
-                    self.assertClose(mn, expectedVal)
+            if scaleByFwhm:
```

Comment:
This suggests to me that you should have two separate test methods, one for `scaleByFwhm=True` and the other for `False`.